.github/workflows/backport.yml

```diff
@@ -1,40 +1,36 @@
-name: Backport changes to all maintenance branches
+name: Backport changes to maintenance branches
 on:
  push:
   branches:
    - main
 jobs:
-  backport:
-    strategy:
-      matrix:
-        branch: ['v/5.0', 'v/5.1', 'v/5.2', 'v/5.3', 'v/5.4', 'v/5.5']
+  get-maintenance-branches:
     runs-on: ubuntu-latest
+    outputs:
+      branches: "['5.3', '5.4', '5.5']"
```

Comment:
Just a suggestion....

The branch names are part of label names so could rely on that and no need for list. Make `gh` call to get pr labels by SHA and traverse over the added backport labels. This will be more exacting then trying to check all labels using `shioyang/check-pr-labels-on-push-action` (makes API calls). I think it should work!

Then no need for separate `get-maintenance-branches` and `backport-to-all-branch` and we can remove from release [instructions](https://hazelcast.atlassian.net/wiki/spaces/EN/pages/4681138242/Create+branch+5.x#Update-hz-docs-backport.yml)

Not saying do it but thought I mention. If good to do then can do it in this or separate PR

Reply:
> The branch names are part of label names so could rely on that and no need for list

We still need a list of maintenance branches for the `all` variant, though - so there's no real saving in configuration.

Extracting from the label is easier to support, though, it's [how I did it in my `hazelcast-mono` backporting workflow](https://github.com/hazelcast/hazelcast-mono/blob/f46b5688dc8c447ac0a4e1d42800b4dce4c2fa33/.github/workflows/backport.yml#L34).
But then you need to do the match-the-commit-with-the-pr-and-get-the-labels calls here instead.

> Make gh call to get pr labels by SHA and traverse over the added backport labels. This will be more exacting then trying to check all labels using shioyang/check-pr-labels-on-push-action (makes API calls). I think it should work!

Not quite sure I follow.
The [`check-pr-labels-on-push-action` uses the GitHub API](https://github.com/shioyang/check-pr-labels-on-push-action/blob/542fcf6f8dbad68c61bcbd4d1bce1531caf31c30/src/index.ts#L3) to get the information, but the `gh` CLI tool does the same - they are both GitHub API clients.

> Then no need for separate get-maintenance-branches and backport-to-all-branch and we can remove from release [instructions](https://hazelcast.atlassian.net/wiki/spaces/EN/pages/4681138242/Create+branch+5.x#Update-hz-docs-backport.yml)

They are different workflows - I'd _guess_ that `all` is for long-lived mistakes (typos etc), where's the selective backporting it for updating documentation on recently introduced features (e.g. UCN - new for `5.4`, so changes need backporting - but not to `5.3`).

Reply:
Understood. May be I was thinking we don't have `all` variant as its easy to add labels individually. Yes will still have to make `gh` (couple) calls but that would be once at the start as opposed to currently we make call for each label in the matrix. I was thinking it will be little more efficient, simpler and also was thinking if we can remove the release post task that would have been good

Anyways, I will approve shortly. Best to delete unwanted labels in GH settings.